core/bin/zksync_server/src/main.rs

```diff
@@ -101,9 +116,42 @@ async fn main() -> anyhow::Result<()> {
         [Component::WitnessGenerator(Some(_), _)]
     );
 
+    // TODO (QIT-22): Only deserialize configs on demand.
+    // Right now, we are trying to deserialize all the configs that may be needed by `zksync_core`.
+    // "May" is the key word here, since some configs are only used by certain component configuration,
+    // hence we are using `Option`s.
+    let configs = TempConfigStore {
```

Comment:
This was required to minimize an (already big) diff size. It's the easiest way to pass the configs to `init_components` without giving it the ability to somehow deserialize configs, so I think it's an OK-ish compromise for a short period of time.